src/ia-book-search-results.js

```diff
@@ -0,0 +1,104 @@
+import { html, LitElement } from 'lit-element';
+import { unsafeHTML } from 'lit-html/directives/unsafe-html';
+import bookSearchResultsCSS from './styles/ia-book-search-results.js';
+import closeIcon from './icon-close.js';
+
+export class IABookSearchResults extends LitElement {
+  static get styles() {
+    return bookSearchResultsCSS;
+  }
+
+  static get properties() {
+    return {
+      results: {
+        type: Array,
+      },
+    };
+  }
+
+  constructor() {
+    super();
+    this.results = [];
+    this.matchRegex = new RegExp('{{{(.+?)}}}', 'g');
+
+    this.bindBookReaderListeners();
+  }
+
+  bindBookReaderListeners() {
+    document.addEventListener('BookReader:SearchCallback', this.setResults.bind(this));
+  }
+
+  setResults({ detail }) {
+    this.results = detail.results;
+  }
+
+  result(match) {
+    const coverImage = html`<img src="${match.cover}" />`;
+    return html`
+      <li>
+        ${match.cover ? coverImage : ''}
+        <h4>${match.title}</h4>
+        ${this.highlightedHit(match.text)}
```

Comment:
_Suggestion_ Since there is some logic going on inside the search result, like the cover image and highlighting, I might pull the search result out to its own component. That would let you encapsulate the result display logic there and relegate the higher-level search functionality to this component.